src/control/server/config.go

```diff
@@ -444,5 +447,86 @@ func (c *Configuration) Validate(log logging.Logger) (err error) {
 			}
 		}
 	}
+
+	if len(c.Servers) > 1 {
+		if err := validateMultiServerConfig(log, c); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// validateMultiServerConfig performs an extra level of validation
+// for multi-server configs. The goal is to ensure that each instance
+// has unique values for resources which cannot be shared (e.g. log files,
+// fabric configurations, PCI devices, etc.)
+func validateMultiServerConfig(log logging.Logger, c *Configuration) error {
+	if len(c.Servers) < 2 {
+		return nil
+	}
+
+	seenValues := make(map[string]int)
+
+	for idx, srv := range c.Servers {
+		fabricConfig := fmt.Sprintf("fabric:%s-%s-%d",
+			srv.Fabric.Provider,
+			srv.Fabric.Interface,
+			srv.Fabric.InterfacePort)
+
+		if seenIn, exists := seenValues[fabricConfig]; exists {
+			log.Debugf("%s in %d duplicates %d", fabricConfig, idx, seenIn)
+			return FaultConfigDuplicateFabric(idx, seenIn)
+		}
+		seenValues[fabricConfig] = idx
+
+		if srv.LogFile != "" {
+			logConfig := fmt.Sprintf("log_file:%s", srv.LogFile)
+			if seenIn, exists := seenValues[logConfig]; exists {
+				log.Debugf("%s in %d duplicates %d", logConfig, idx, seenIn)
+				return FaultConfigDuplicateLogFile(idx, seenIn)
+			}
+			seenValues[logConfig] = idx
+		}
+
+		scmConf := srv.Storage.SCM
+		mountConfig := fmt.Sprintf("scm_mount:%s", scmConf.MountPoint)
+		if seenIn, exists := seenValues[mountConfig]; exists {
+			log.Debugf("%s in %d duplicates %d", mountConfig, idx, seenIn)
+			return FaultConfigDuplicateScmMount(idx, seenIn)
+		}
+		seenValues[mountConfig] = idx
+
+		if len(scmConf.DeviceList) > 0 {
+			devStr := fmt.Sprintf("scm_list:%s", sliceToString(scmConf.DeviceList))
+
+			if seenIn, exists := seenValues[devStr]; exists {
+				log.Debugf("%s in %d duplicates %d", devStr, idx, seenIn)
+				return FaultConfigDuplicateScmDeviceList(idx, seenIn)
+			}
+			seenValues[devStr] = idx
+		}
+
+		bdevConf := srv.Storage.Bdev
+		if len(bdevConf.DeviceList) > 0 {
+			devStr := fmt.Sprintf("bdev_list:%s", sliceToString(bdevConf.DeviceList))
+
+			if seenIn, exists := seenValues[devStr]; exists {
+				log.Debugf("%s in %d duplicates %d", devStr, idx, seenIn)
+				return FaultConfigDuplicateBdevDeviceList(idx, seenIn)
+			}
+			seenValues[devStr] = idx
+		}
+	}
+
 	return nil
 }
+
+// sliceToString converts a string slice to a stable string
+// representation
+func sliceToString(in []string) string {
+	out := make([]string, len(in))
+	copy(out, in)
```

Comment:
why not mutate input?

Reply:
Just safer this way, in case there is a reason for the input slice to have a specific ordering. It's generally good practice to avoid side-effects. The point of the function is to convert a slice into a stable string representation regardless of input order.